packages/app/src/systems/Asset/components/AssetInput/AmountInput.tsx

```diff
@@ -0,0 +1,100 @@
+import { cssObj } from '@fuel-ui/css';
+import { Button, Flex, Input, Text } from '@fuel-ui/react';
+import type { BN } from 'fuels';
+import { useState } from 'react';
+import type { FC, ChangeEvent } from 'react';
+
+import { AmountInputLoader } from './AmountInputLoader';
+
+export type AmountInputProps = {
+  amount: BN;
+};
+
+type AmountInputComponent = FC<AmountInputProps> & {
+  Loader: typeof AmountInputLoader;
+};
+
+export const AmountInput: AmountInputComponent = ({ amount }) => {
+  const [assetAmount, setAssetAmount] = useState<string>();
+  const formatArgs = {
+    minPrecision: 2,
```

Comment:
should include 
`precision: DECIMAL_UNITS`,


otherwise more than 3 precision values won't be showed correctly